perfkitbenchmarker/flag_util.py

```diff
@@ -219,3 +220,49 @@ def __enter__(self):
   def __exit__(self, *unused_args, **unused_kwargs):
     """Stops the flag substitution."""
     self._flags.__dict__['FlagDict'] = self._original_flagdict
+
+
+class UnitsParser(flags.ArgumentParser):
+  """Parse a flag containing a unit expression."""
+
+  syntactic_help = 'A quantity with a unit. Ex: 12MB'
+
+  def Parse(self, inp):
+    """Parse the input.
+
+    Args:
+      inp: a string
+
+    Returns:
+      A pint.Quantity.
+
+    Raises:
+      ValueError if it can't parse its input.
+    """
+
+    if isinstance(inp, pint_wrapper.ureg.Quantity):
+      return inp
+
+    try:
+      return pint_wrapper.ureg.parse_expression(inp)
+    except Exception as e:
+      raise ValueError("Couldn't parse unit expresion %s:" % inp,
+                       e.message)
+
+
+class UnitsSerializer(flags.ArgumentSerializer):
+  def Serialize(self, units):
+    return str(units)
+
+
+def DEFINE_units(name, default, help,
+                 flag_values=flags.FLAGS, **args):
+  """Register a flag whose value is a units expression."""
```

Comment:
Documented the arguments.

I'm not sure I can do the `name, *args, **kwargs` thing because I inject my own `serializer` argument in position 5 of the positional arguments. I could modify the `*args` tuple, but that seems pretty convoluted.